packages/react-components/react-switch/src/components/Switch/useSwitchStyles.ts

```diff
@@ -184,6 +184,32 @@ const useInputBaseClassName = makeResetStyles({
         color: 'GrayText',
       },
     },
+
+    ':enabled:checked': {
+      ':hover': {
+        [`& ~ .${switchClassNames.indicator}`]: {
+          backgroundColor: 'Highlight',
+          color: 'Canvas',
+        },
+      },
+      [`& ~ .${switchClassNames.indicator}`]: {
+        backgroundColor: 'Highlight',
+        color: 'Canvas',
+      },
+    },
+
+    ':disabled:checked': {
```

Comment:
The enabled checked state change looks good but the disabled one doesn't align with the design spec. We should keep the implementation and design in sync so one or the other needs to be changed.

![switch_design_spec](https://user-images.githubusercontent.com/93940821/226401012-8b27e51f-40cc-44ad-a23e-719b9e62b4c5.png)

Reply:
For what it's worth, that spec screenshot is showing the "Teams High Contrast" theme, which is different from forced-colors high contrast on Windows that this PR is changing. However, it's still worth checking with design about what the forced-colors visuals should be.

If you want to get this PR in sooner than later, you could remove the `:disabled:checked` styles for now, and make a follow-up PR with them once you hear back from design.